From 608d0f1caf43628c8260a994c277a040f1e616e0 Mon Sep 17 00:00:00 2001
From: Peter Bergin <peter.bergin@windriver.com>
Date: Tue, 23 May 2023 13:19:55 +0200
Subject: [PATCH] CMakeLists.txt: fix substitution for static libs

During configuration the liibrary names are substitued for
the static build. When building with tls support the first
substitution will match both cases and trash the name for hiredis_ssl.

Input string: 'hiredis::hiredis;hiredis::hiredis_ssl;'
Output string: 'hiredis::hiredis_static;hiredis::hiredis_static_ssl;'
Wanted string: 'hiredis::hiredis_static;hiredis::hiredis_ssl_static;'

Fix string replacement by also including the ';' token to better match.

Upstream-Status: Submitted [https://github.com/sewenew/redis-plus-plus/issues/486]

Signed-off-by: Peter Bergin <peter.bergin@windriver.com>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 23a1840..5478271 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -184,13 +184,13 @@ if(REDIS_PLUS_PLUS_BUILD_STATIC)
     set(REDIS_PLUS_PLUS_HIREDIS_LIBS_STATIC ${REDIS_PLUS_PLUS_HIREDIS_LIBS})
 
     if (TARGET hiredis::hiredis_static)
-      string(REPLACE "hiredis::hiredis" "hiredis::hiredis_static" REDIS_PLUS_PLUS_HIREDIS_LIBS_STATIC
+      string(REPLACE "hiredis::hiredis;" "hiredis::hiredis_static;" REDIS_PLUS_PLUS_HIREDIS_LIBS_STATIC
         "${REDIS_PLUS_PLUS_HIREDIS_LIBS_STATIC}")
     endif()
 
     if (TARGET hiredis::hiredis_ssl_static)
       # If SSL is not enabled, this line will have no effect
-      string(REPLACE "hiredis::hiredis_static_ssl" "hiredis::hiredis_ssl_static" REDIS_PLUS_PLUS_HIREDIS_LIBS_STATIC
+      string(REPLACE "hiredis::hiredis_static_ssl;" "hiredis::hiredis_ssl_static;" REDIS_PLUS_PLUS_HIREDIS_LIBS_STATIC
         "${REDIS_PLUS_PLUS_HIREDIS_LIBS_STATIC}")
     endif()
 
-- 
2.34.1

